syncmaster/scheduler/settings/__init__.py

```diff
@@ -0,0 +1,28 @@
+# SPDX-FileCopyrightText: 2023-2024 MTS PJSC
+# SPDX-License-Identifier: Apache-2.0
+from pydantic import Field
+
+from syncmaster.settings import SyncmasterSettings
+
+
+class SchedulerSettings(SyncmasterSettings):
+    """Celery scheduler settings.
+
+    Examples
+    --------
+
+    .. code-block:: bash
+
+        SYNCMASTER__SCHEDULER__TRANSFER_FETCHING_TIMEOUT=200
```

Comment:
Have you checked if scheduler really uses `SYNCMASTER__SCHEDULER__TRANSFER_FETCHING_TIMEOUT` or the setting key is actually `SYNCMASTER__TRANSFER_FETCHING_TIMEOUT`?

Reply:
fixed docstring
the settings key is `SYNCMASTER__SCHEDULER__TRANSFER_FETCHING_TIMEOUT_SECONDS` (because of alias) and checked that it is applied

Reply:
> checked that it is applied

Could you give an example? I don't see where `SYNCMASTER__SCHEDULER__` prefix comes from

Reply:
there is no prefix, just hardcoded alias in pydantic field 

```
TRANSFER_FETCHING_TIMEOUT_SECONDS: int = Field(
        180,
        description="Timeout for fetching transfers in seconds",
        alias="SYNCMASTER__SCHEDULER__TRANSFER_FETCHING_TIMEOUT_SECONDS",
    )
```

Reply:
![image](https://github.com/user-attachments/assets/b0d690f1-0abb-4a8f-9a71-a99168b085a1)
]

Reply:
So if you want to build custom Settings object, and pass it to worker factory, you have to use `SchedulerSettings(SYNCMASTER__SCHEDULER__TRANSFER_FETCHING_TIMEOUT_SECONDS=180)` instead of `SchedulerSettings(TRANSFER_FETCHING_TIMEOUT_SECONDS=180)`?

Reply:
yes, I didn't find another approach to redefine prefix for parameters related to worker setting with keeping same prefix for common parameters like database, broker

Reply:
My recommendation is to use Data.Rentgen or Horizon approach:

```python
class ServerAppSettings(PydanticSettings):
  database: DatabaseSettings
  broker: BrokerSettings
  auth: AuthSettings
  server: ServerSpecificSettings

  class Config:
    prefix = "SYNCMASTER__"

class WorkerAppSettings(PydanticSettings):
  database: DatabaseSettings
  broker: BrokerSettings
  worker: WorkerSpecificSettings

  class Config:
    prefix = "SYNCMASTER__"

class SchedulerAppSettings(PydanticSettings):
  database: DatabaseSettings
  broker: BrokerSettings
  scheduler: SchedulerSpecificSettings

  class Config:
    prefix = "SYNCMASTER__"
```

Advantages:
* You can programmatically create settings like `ServerAppSettings(database=DatabaseSettings(...), server=ServerSpecificSettings(...))` without any prefixes created by `alias=...`
* You can set the same variable `SYNCMASTER__DATABASE__URL=...`, and it will be converted to `settings.database.url=...` by pydantic settings, without any app-related prefix.
* Application-specific settings have a prefix, like `SYNCMASTER__WORKER__CREATE_SPARK_FUNCTION` (`SYNCMASTER__` prefix in PydanticSettings config + field name of `worker` nested config + `CREATE_SPARK_FUNCTION` field of worker-specific config.
* All setting classes are independent, you can create `ServerAppSettings()` without worker-specific settings, or `WorkerAppSettings()` without auth-specific settings.

Reply:
settings are splited into `ServerAppSettings`, `WorkerAppSettings`, `SchedulerSettings`